### ISSUE TITLE
【FR】Bug: Operator crash on 华为(Ascend) (Alibaba-NLP/Tongyi-DeepResearch-30B-A3B)

### ISSUE THREAD

**Issue by Lxiparer** [FlagOpen/FlagGems#6270]
### Description

FlagGems 的 `linear` / `softmax` 两个算子在 **Ascend 910** 上输出全 `!` 乱码，导致模型服务虽然进程存活，但生成内容不可用（先输出正常、随后退化为 `!!!`，间歇性复现）。

vllm-plugin-FL 随包配置 `vllm_fl/dispatch/config/ascend.yaml` 的 `flagos_blacklist` 段落已记录该缺陷（原文）：

```
    # --- Precision: linear (KernelGen Nvidia GEMM) outputs garbage on Ascend 910 -> all '!' ---
    - linear
    - softmax
```

`flag_gems/ops/linear.py` 文件头声明该实现由 KernelGen 生成（`# Generated by KernelGen: https://github.com/flagos-ai/KernelGen`），即 Nvidia 侧 GEMM kernel 在 Ascend 后端未正确适配。

**规避**：将 `linear`、`softmax` 纳入 `VLLM_FL_FLAGOS_BLACKLIST`（本模型固化为 39 算子并集），改用非 FlagGems 实现后输出恢复正常。

### Environment

| Item | Value |
|------|-------|
| Hardware | 华为(Ascend) |
| GPU | 910C |
| PyTorch | 2.10.0+cpu |
| FlagGems | 5.3.4.post1.dev128+g05cd332b3 |
| vLLM | 0.20.2 |

### Steps to Reproduce

1. Set up environment on 华为(Ascend)
2. Install dependencies (PyTorch + Triton + FlagGems + vLLM)
3. Start vLLM inference service with FlagGems enabled
4. Observe operator runtime failures / framework errors

### Expected Behavior

All operators should execute correctly and inference should complete without errors.

### Actual Behavior

FlagGems `linear` / `softmax` 被启用后，模型推理输出**先正常、随后退化为全 `!` 乱码**（间歇性复现）；服务进程未崩溃，但输出不可用，等价于服务不可用。

- **触发**：`--mode flagos` 首启走代码注入全量开启 FlagGems 算子，绕过 `ascend.yaml` 的 `flagos_blacklist`
- **现象**：`Coherent start then degrades to "!!!" — the classic FlagGems garbage-output signature on Ascend`
- **恢复**：将 `linear` / `softmax` 等并入 `VLLM_FL_FLAGOS_BLACKLIST`（本模型固化 39 算子并集）后，输出恢复为正常（`Paris` / `180`）

### Error Logs

```
[Tongyi-DeepResearch-30B-A3B] FlagGems linear/softmax 在 Ascend 910 输出全 '!' 乱码
证据摘录（逐字取自原始日志，[from:] 标注来源）

[from: logs/issues_startup.log:1-4]
[2026-09-09 07:23:00] V2(flagos) | plugin首启部分算子输出全! 乱码（间歇性）
  详情: --mode flagos 首启走代码注入全量开启，绕过 ascend.yaml flagos_blacklist，linear/softmax 等 KernelGen kernel 在 Ascend910 输出乱码
  操作: persist_op_config 固化 VLLM_FL_PREFER_ENABLED=true + 39算子 ascend.yaml blacklist 并集到 /etc/environment，plugin dispatch 接管
  结果: 恢复成功，24算子生效，推理输出正常(Paris/180)，service_ok=true

[from: logs/terminal.log:35]
Coherent start then degrades to "!!!" - the classic FlagGems garbage-output signature on Ascend plugin first-boot.

[from: logs/pipeline.log:22]
[2026-09-09 15:25:11] 首启乱码 (linear/softmax) -> 固化 39 算子 ascend.yaml 黑名单并集 + plugin dispatch -> 推理正常

[from: vllm-plugin-FL 随包配置 vllm_fl/dispatch/config/ascend.yaml flagos_blacklist 段]
    # --- Precision: linear (KernelGen Nvidia GEMM) outputs garbage on Ascend 910 -> all '!' ---
    - linear
    - softmax

[FlagGems 集成方式]
  FLAGGEMS_SOURCE_DIR=/workspace/FlagGems/src/flag_gems
  enable_call: flag_gems.only_enable(record=False, include=FLAG_GEMS_CONFIG)
  集成文件: /vllm-workspace/vllm/vllm/model_executor/models/minicpmo.py:79 import flag_gems
            /vllm-workspace/vllm/vllm/model_executor/models/minicpmo.py:117 flag_gems.only_enable(record=False, include=FLAG_GEMS_CONFIG)

[FlagGems 源码实证]
  flag_gems/ops/linear.py 文件头: # Generated by KernelGen: https://github.com/flagos-ai/KernelGen
```

### FlagGems Integration Code

**FlagGems 源码目录**：`/workspace/FlagGems/src/flag_gems`

**enable 调用**：

```python
flag_gems.only_enable(record=False, include=FLAG_GEMS_CONFIG)
```

**集成位置**：`/vllm-workspace/vllm/vllm/model_executor/models/minicpmo.py`

```
:79    import flag_gems
:117   flag_gems.only_enable(record=False, include=FLAG_GEMS_CONFIG)
```

**相关算子源码**：`flag_gems/ops/linear.py`（文件头 `# Generated by KernelGen: https://github.com/flagos-ai/KernelGen`）、`softmax`

### Additional Context

- The issue appears to be specific to **华为(Ascend)** platform and may not reproduce on standard CUDA (NVIDIA) environments.

### Possible Directions

- Verify kernel support for the failing operators on this backend
- Check dtype compatibility (e.g., BF16 / FP16)
- Investigate Triton code generation for this platform
- Confirm whether fallback kernels are correctly triggered
